### ISSUE TITLE
VNL5090 specifications are not guaranteed with 3.3V supply

### ISSUE THREAD

**Issue by dron0gus** [FOME-Tech/Polygonus#27]
![Screenshot from 2024-01-28 19-56-09](https://github.com/FOME-Tech/Polygonus/assets/28624689/d5060e6c-59a4-4e05-9afc-8b4857fde7a8)

SO-8 version has dedicated supply voltage pin, BTW.

**Comment by rusefillc** [comment]
A bit related to https://github.com/mck1117/proteus/issues/125

**Comment by OrchardPerformance** [comment]
Spent some time running tests on the VNL5090 all seem to trigger by 2.7v and we reliably see 3.2v on the brain boards at the VNL pin. 

No one has yet had an issue with the lowsides and there must be 100 cars and over 50,000 miles driven. 

Probably on a V2 of the brain board we will do something to bump the voltage up but it currently seems to be a spec vs reality thing.

**Comment by OrchardPerformance** [comment]
What temperature conditions would you like? 
We can do it at anything from the 15c it is here today right up to 100c if you want?

**Comment by nmschulte** [comment]
I wonder how the supply affects switching characteristics: that's probably where an issue would arise if any, given your tests @ 3.3V.

**Comment by OrchardPerformance** [comment]
Yes, that would certainly be my primary concern. In terms of what it would directly affect the most obvious would be injector dead times. Something that can certainly be verified using a spare rover 4cu ecu as they have VNL5090s on the base board and 5v pull ups. 
A job for a dedicated test on that at a later time.

**Comment by OrchardPerformance** [comment]
Repeated the same testing on the same VNL5090s today at 70c PCB temperature, increased temperature actually lowers the activation voltage. At 70c the activation is 0.1v lower than at 15c.

**Comment by dron0gus** [comment]
As far as I remember I already posted these pictures in RE slack channel related to WBO. This is how similar part (VND7NV04) part operates with 2.7V vs 3.1V and 1A load.
![20221213_003334](https://github.com/FOME-Tech/Polygonus/assets/28624689/8fe13126-db01-45c5-8f27-d357efe432b7)
![20221213_003511](https://github.com/FOME-Tech/Polygonus/assets/28624689/3f2331d-f7f2-4105-ac75-58006ceca6b1)
You may see significant voltage drop on this MOSFET when supplied from 2.7V. I see 1V drop, with only 1A load. For me it seems that MOSFET is far out of its switching region.
While DS says that maximum threshold voltage is 2.5V, Rds vs Vin graph starts at 3.5V. And it starts from 110mOhm@150C which is quite much.
3.1V amplitude was reduced to 2.7V by simply changing output pin "Output speed setting" from "Very high speed" to "Low speed" for STM32.
BTW STM32F4 is capable to keep working at wide supply range. Down to 1.7V if I recall correctly.

**Comment by dron0gus** [comment]
![20221213_003511](https://github.com/FOME-Tech/Polygonus/assets/28624689/104991a4-38d7-4c9d-b978-c63d3aeec01a)
This one is for 2.7V.

**Comment by dron0gus** [comment]
Also, please note that transition process is bit extended with lower drive voltage. This may affect injector dead time calculation mentioned above. And cause additional heating of MOSFET

**Comment by OrchardPerformance** [comment]
That is exactly what my main concern would be and the reason for making V2 work off 5v. 

I'll pop this one back open for now and take some measurements of the resistance and on time of the VNL5090 actually on brain board. 
It would be good to know with respect to the injector on time effects even if it does not appear to directly be causing an issue.
